### PR TITLE
Add shared library target to CMake

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -52,6 +51,11 @@ else()
     add_compile_options(-g)
 endif()
 
+option(BUILD_SHARED "Build a shared library and CLI too" OFF)
```
Comment (bnoordhuis) [inline CMakeLists.txt]: This should use [BUILD_SHARED_LIBS](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html#variable:BUILD_SHARED_LIBS), I think?

Comment (saghul) [reply CMakeLists.txt]: Oh that will simplify things, TIL!

**Comment by saghul** [comment]
Updated!

**Review by bnoordhuis — APPROVED** [review]
Nice, LGTM.

**Comment by bnoordhuis** [comment]
...but needs a s/qjs-shared/qjs/g :)

**Comment by theoparis** [comment]
This seems to be broken now. I get a bunch of errors like `ld.lld: error: undefined symbol: js_std_add_helpers` with `-DBUILD_SHARED_LIBS=On`.

**Comment by saghul** [comment]
Can you share the full build output?

**Pull request merged**